charon/tests/ui/issue-45-misc.out

```diff
@@ -144,6 +144,8 @@ trait core::iter::traits::iterator::Iterator<Self>
 }
 
 trait core::iter::traits::collect::IntoIterator<Self>
+where
+    (Self::IntoIter::[@TraitClause0])::Item = Self::Item,
```

Comment:
This change is because we used to do `ctx.get_predicates()` before exploring trait associated items, and we never tracked item-specific predicates so we forgot this one.